File: tests/v11/fixtures/rebuilt-out/bridges/typescript/src/a4_sy_orchestration/bridge_main.ts
import type { SpawnSyncReturns } from "node:child_process";

import type { BridgeManifest } from "../a0_qk_constants/bridge_contract.js";
import { PythonBridgeClient } from "../a2_mo_composites/python_bridge_client.js";
import { describeBridge } from "../a3_og_features/bridge_feature.js";

export function loadBridgeSummary(root: string = process.cwd()): BridgeManifest {
  return describeBridge(root);
}

export function runAssAde(args: string[], root: string = process.cwd()): SpawnSyncReturns<string> {
  return new PythonBridgeClient(root).run(args);
}

File: tests/fixtures/rebuilt-out/bridges/rust/src/a4_sy_orchestration/bridge_main.rs
use std::env;

use crate::a0_qk_constants::bridge_contract::BridgeManifestSummary;
use crate::a2_mo_composites::python_bridge_client::PythonBridgeClient;
use crate::a3_og_features::bridge_feature::describe_bridge;

pub fn load_bridge_summary() -> Result<BridgeManifestSummary, String> {
    describe_bridge(env::current_dir().map_err(|err| err.to_string())?)
}

pub fn run() -> Result<(), Box<dyn std::error::Error>> {
    let client = PythonBridgeClient::new(env::current_dir()?).map_err(std::io::Error::other)?;
    let payload = client.manifest_payload().map_err(std::io::Error::other)?;
    println!("{}", payload);
    Ok(())
}
